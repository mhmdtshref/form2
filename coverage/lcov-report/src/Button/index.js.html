<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/Button/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/Button</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react'
&nbsp;
class Button extends Component {
  constructor(props) {
    super(props);
  }
&nbsp;
  generateProps = (props) =&gt; {
    let newProps = {...props};
    delete newProps.onChange;
    delete newProps.formstate;
    return newProps;
  };
&nbsp;
  render() {
    return &lt;button {...this.generateProps(this.props)}&gt;{ this.props.children || this.props.value }&lt;/button&gt;;
  }
}
&nbsp;
export default Button;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jul 09 2019 12:33:04 GMT+0300 (Eastern European Summer Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
