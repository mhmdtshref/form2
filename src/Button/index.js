import React, { Component } from 'react'

class Button extends Component {
  constructor(props) {
    super(props);
  }

  generateProps = (props) => {
    let newProps = {...props};
    delete newProps.onChange;
    delete newProps.formstate;
    return newProps;
  };

  render() {
    return <button {...this.generateProps(this.props)}>{ this.props.children || this.props.value }</button>;
  }
}

export default Button;
