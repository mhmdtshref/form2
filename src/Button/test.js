import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index';

describe('Button Field Test', () => {

  it('is truthy', () => {
    expect(Button).toBeTruthy()
  });

  it('Test natural button', () => {
    const button = renderer.create(<Button name="button" />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Test button with value using value property', () => {
    const button = renderer.create(<Button name="button" value="Click me!" />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Test button with value using inner text value', () => {
    const button = renderer.create(<Button name="button">Click me!</Button>).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Test button with onClick function property', () => {
    const onClickHandler = () => {
      // eslint-disable-next-line no-console
      console.log("Test console log!");
    };
    const button = renderer.create(<Button name="button" onClick={onClickHandler} />).toJSON();
    expect(button).toMatchSnapshot();
  });
});
