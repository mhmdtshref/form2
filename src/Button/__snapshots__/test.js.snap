// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Field Test Test button with onClick function property 1`] = `
<button
  name="button"
  onClick={[Function]}
/>
`;

exports[`Button Field Test Test button with value using inner text value 1`] = `
<button
  name="button"
>
  Click me!
</button>
`;

exports[`Button Field Test Test button with value using value property 1`] = `
<button
  name="button"
  value="Click me!"
>
  Click me!
</button>
`;

exports[`Button Field Test Test natural button 1`] = `
<button
  name="button"
/>
`;
